### DISCUSSION TITLE
Cannot suppress DestructuringDeclarationWithTooManyEntries

### DISCUSSION THREAD

**Discussion by TWiStErRob** [detekt/detekt#5123] [Q&A]
```kotlin
val list = listOf("a", "b", "c", "d")

val (a, b, c, d) = list // DestructuringDeclarationWithTooManyEntries

println("$a,$b,$c,$d")
```

now, that detection is valid, and I want to choose to ignore it
```kotlin
val list = listOf("a", "b", "c", "d")

@Suppress("DestructuringDeclarationWithTooManyEntries")
val (a, b, c, d) = list

println("$a,$b,$c,$d")
```

but it just gives me the finger:
![image](https://user-images.githubusercontent.com/2906988/180611285-2f381599-2f9c-42f3-801a-845db4e0f7c4.png)

another try:
```kotlin
val (a, b, c, d) = @Suppress("DestructuringDeclarationWithTooManyEntries") list
```
![image](https://user-images.githubusercontent.com/2906988/180611347-a16356c1-bc35-4316-8957-9dfc6794b741.png)

this is the point where I give up:
![image](https://user-images.githubusercontent.com/2906988/180611442-89194650-ea76-4fae-b1ff-8b77cb0a3ab5.png)

This kind-of works, but laughably pointless because it needlessly scopes the variables:
```kotlin
run @Suppress("DestructuringDeclarationWithTooManyEntries") {
    val (a, b, c, d) = list
}
```

**Comment by cortinico** [comment]
Have you tried suppressing the containing block (like a function or class)?

**Reply by TWiStErRob** [reply]
That works of course, it just goes against the principle of least scope, and affects way more than necessary. Just because one destructuring is fine in a function, it doesn't mean the next one written months later by another developer is also fine.

`<storytime>`
This is one thing I really love about Kotlin, we can put annotations on expressions. A really going all the way example:
```kotlin
private val x = y * @Suppress("MagicNumber") 127
```
Yes, I could put it on `val x`, but then someone comes about and adds `x = y * 127 + 1234` and that 1234 gets automatically suppressed, and Detekt wouldn't flag the *new* MagicNumber.

This is a contrived and edgy example, but it shows how much control we have and how many implicit decisions we make (for someone else in the future) just when deciding placement of `@Suppress`.
`</storytime>`

I think it's fair to ask to be able to put the `Suppress` on the violating line. You might remember how we moved the `@Suppress`-ability from `try` to "`catch` variable" for another rule.

I also understand this is a Kotlin limitation (don't understand why destructuring is not an expression though...), and we could decide on a won't fix, *but* maybe there's a way I just didn't think of, or we can change the rule to read the suppress from the RHS maybe, or a similar approach.

**Reply by cortinico** [reply]
> I also understand this is a Kotlin limitation (don't understand why destructuring is not an expression though...)

I haven't looked into this deeply, but I believe this is the case. I'm not sure why you can't apply the `@Suppress` annotation to destructuring declaration, that should be investigated. If it's a Kotlin limitation, you should open an issue on YouTrack and as a workaround apply the annotation to the nearest block.
